synapse/lib/storm.py

```diff
@@ -783,6 +784,83 @@
     },
 )
 
+class DmonManager(s_base.Base):
+    '''
+    Manager for StormDmon objects.
+    '''
+    async def __anit__(self, core):
+        await s_base.Base.__anit__(self)
+        self.core = core
+        self.dmons = {}
+        self.enabled = False
+
+        self.onfini(self._finiAllDmons)
+
+    async def _finiAllDmons(self):
+        await asyncio.gather(*[dmon.fini() for dmon in self.dmons.values()])
+
+    async def _stopAllDmons(self):
+        await asyncio.gather(*[dmon.stop() for dmon in self.dmons.values()])
+
+    async def addDmon(self, iden, ddef):
+        dmon = await StormDmon.anit(self.core, iden, ddef)
+        self.dmons[iden] = dmon
+        # TODO Remove default=True when dmon enabled CRUD is implemented
+        if self.enabled and ddef.get('enabled', True):
+            await dmon.run()
+        return dmon
+
+    def getDmonRunlog(self, iden):
+        dmon = self.dmons.get(iden)
+        if dmon is not None:
+            return dmon._getRunLog()
+        return ()
+
+    def getDmon(self, iden):
+        return self.dmons.get(iden)
+
+    def getDmonPacked(self, iden):
```

Comment:
these have been plumbed as Def / Defs APIs in other things.  getDmonDef / getDmonDefs would be perfect here